common/lib/protocol-definitions/src/protocol.ts

```diff
@@ -312,6 +312,17 @@ export interface ISignalMessage {
 	// eslint-disable-next-line @rushstack/no-new-null
 	clientId: string | null;
 
+	/**
+	* The source of the signal. This can be a GUID of the client,
+	* "server" for server generated signals and "external" for
+	* services that use the new Fluid Service broadcast-signal API
+	* to relay signals through the server. To maintain
+	* backward-compat for now, we are making it optional. Over
+	* time clientId should be phased out of use and signalSource
+	* should become the main source of truth.
+	*/
+   signalSource?: string;
```

Comment:
Since there are special values the system will look for, it might be worth adding them at the type level (enhances discoverability of the special values, and gives us a place to document them).

E.g.
```suggestion
   signalSource?: "external" | "server" | string;
```

I would further recommend creating a type alias for this to be the home for documentation at the type level.